### ISSUE TITLE
Using Decorators for CSS Module Binding, ES6 vs 7

### ISSUE THREAD

**Issue by andytango** [stylesuxx/generator-react-webpack-redux#62]
Hey, I've noticed that the generator tries to bind css modules to components using ES7 decorators:

```javascript
@cssmodules(styles)
class SomeComponent extends React.Component {
  // ...
}
```

This is a nice syntax, but it doesn't seem to work with the scaffold I created with v4.4.1 of this generator. The styles aren't applied, and React complains that the styleName prop is unknown.

I don't know if there's been an update to ensure full ES7 support since then? 

In any case, I can easily work around the problem by binding the cssmodule in the traditional way:

```javascript
class SomeComponent extends React.Component {
  // ...
}
export default cssmodules(SomeComponent, styles);
```

... but it's a bit of a pain, and I'm not that keen on changing my webpack/babel config to support ES7 when I don't really need/want those features on this project.

I'm therefore wondering if a solution could be to add a command line flag to the all of the generators that suppresses ES7 features and only outputs ES6 compliant code? 

I'd be happy to do a PR for this, but I think it might have quite a large impact (would need to switch templates in various places?), so this might need some discussion?

**Comment by keithriver** [comment]
Seems that decorators aren't included in the default configuration (strangely so, since they're used in subgenerators). In fact by default there is only an `es2015` preset. Technically you can install legacy plugins like `babel-plugin-transform-decorators-legacy` and `transform-class-properties` but it thinks it's already time to do just that:

`npm i --save-dev babel-preset-es2016`

```javascript
    // ./conf/webpack/Base.js
    ...
    rules: [
          {
            enforce: 'pre',
            test: /\.js?$/,
            include: this.srcPathAbsolute,
            loader: 'babel-loader',
            query: {
              presets: ['es2016'] // ['stage-0', 'react'],
              // plugins: ['react-html-attrs', 'transform-decorators-legacy', 'transform-class-properties']
            }
          },
    ...
```

**Comment by stylesuxx** [comment]
Hey guys,
interesting, this seems to be an issue in the base  gen => https://github.com/react-webpack-generators/generator-react-webpack/blob/v4.0.1-1/generators/component/templates/4/components/Base.js

This is the template used for component generation, I am not entirely sure, but this might be the culprit. I will investigate further tomorrow.

Also webpack config is handled by the base generator.

**Comment by stylesuxx** [comment]
I was using the latest v4 branch of the base generator, which had the decorator in one of the templates. Switched to master branch of the base generator, since this is the base for the next v4 version where this issue is fixed.

**Comment by stylesuxx** [comment]
Fixed in 1.1.0 - closing down here.

For webpack related discussion please go to the [template repository](https://github.com/react-webpack-generators/react-webpack-template).
For component related discussion please go to the [base generator repositroy](https://github.com/react-webpack-generators/generator-react-webpack).
